developing-without-a-build/article.md

```diff
@@ -0,0 +1,257 @@
+
+However nowadays I think build tools and bundlers should be primarily for production optimization, for development I think we can go back to basics. Browser support for es module features has improved a lot in recent years.
+
+In this article we will explore different ways of loading modules in the browser, and offer practical advice on how to handle different scenarios. We will start by looking at javascript modules, and later look at html, css and json as well.
+
+## Loading modules in the browser
+This is not exact step by step tutorial, but you can follow along with any of the examples by using any simple web server. For example `http-server` from npm. Run it with `-c-1` to disable time based caching.
+
+```bash
+npm i -D http-server
+npx http-server -o -c-1
+```
+
+### Loading modules
+Modules are loaded in the browser using regular script tags with a `type="module"` attribute. We can just write our module code directly inline:
+```html
+<!DOCTYPE html>
+<html>
+<head></head>
+
+<body>
+  <script type="module">
+    console.log('hello world!');
+  </script>
+</body>
+
+</html>
+```
+
+From here we can use static imports to load other modules:
+```html
+<script type="module">
+  import './app.js';
+
+  console.log('hello world!');
+</script>
+```
+
+Note that we need to use an explicit file extension, as the browser doesn't know which file to request otherwise.
+
+The same thing works if we use the `src` attribute:
+
+```html
+<script type="module" src="./app.js"></script>
+```
+
+### Loading dependencies
+We don't write our code in just one file. After importing the initial module, we can import other modules. For example let's create two new files:
+
+`src/app.js`:
+```javascript
+import { message } from './message.js';
+
+console.log(`The message is: ${message}`);
+```
+
+`src/message.js`:
+```js
+export const message = 'hello world';
+```
+
+Place both files in a `src` directory and import `app.js` from your index.html:
+
+```html
+<!DOCTYPE html>
+<html>
+<head></head>
+
+<body>
+  <script type="module" src="./src/app.js"></script>
+</body>
+
+</html>
+```
+
+If you run this and check the network panel, you will see both modules being loaded. Because imports are resolved relatively, `app.js` can refer to `message.js` using a relative path:
+
+![](./images/modules.png)
+
+This seems trivial, but it is extremely useful and something we did not have before with classic scripts. We no longer need to coordinate dependencies somewhere central or maintain a base url. Modules can declare their own dependencies, and we can import any module without knowing what their dependencies are. The browser takes care of requesting the correct files.
+
+### Dynamic imports
```

Comment:
You should note that dynamic import()s, since they initiate their own module graph, can be used even in scripts

```html
<script>
  import('./message.js')
    .then(({ message }) => message)
    .then(console.log)
</script>
```